src/url.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+use rquickjs::{
+    module::{Declarations, Exports, ModuleDef},
+    Class, Ctx, Result,
+};
+
+use crate::{
+    http::{url::URL, url_search_params::URLSearchParams},
+    module::export_default,
+};
+pub struct UrlModule;
+
+impl ModuleDef for UrlModule {
+    fn declare(declare: &mut Declarations) -> Result<()> {
+        declare.declare("URL")?;
+        declare.declare("URLSearchParams")?;
+        declare.declare("default")?;
+        Ok(())
+    }
+
+    fn evaluate<'js>(ctx: &Ctx<'js>, exports: &mut Exports<'js>) -> Result<()> {
+        let url_constructor =
+            Class::<URL>::create_constructor(ctx)?.expect("Could note create URL constructor");
```

Comment:
Use  `Class::<URL>::define(default)?;` and ` Class::<URLSearchParams>::define(default)?;` inside export default instead of manually creating constructors.